internal/incident/incident.go

```diff
@@ -575,6 +640,115 @@ func (i *Incident) RestoreEscalationStateRules(states []*EscalationState) {
 	}
 }
 
+// RetriggerEscalations tries to re-evaluate the escalations and notify contacts.
+func (i *Incident) RetriggerEscalations() {
+	i.Lock()
+	defer i.Unlock()
+
+	i.runtimeConfig.RLock()
+	defer i.runtimeConfig.RUnlock()
+	defer i.timer.Done()
+
+	if !i.RecoveredAt.IsZero() {
+		// Incident is recovered in the meantime.
+		return
+	}
+
+	ctx := context.TODO()
+	if i.RestoreRecipients(ctx) != nil {
+		// The error is already logged within `i.RestoreRecipients()`
+		return
+	}
+
+	i.evaluateEscalations(i.StartedAt.Add(i.timer.incidentAgeFilter))
```

Comment:
Make that function return whether any new escalations were triggered. If there were none, you can return here and not send any notifications.